swarm/network/fetcher.go

```diff
@@ -32,6 +32,8 @@ var searchTimeout = 1 * time.Second
 // Also used in stream delivery.
 var RequestTimeout = 10 * time.Second
 
+var maxHopCount uint64 = 20 // maximum number of forwarded requests (hops), to make sure requests are not forwarded forever in peer loops
```

Comment:
actually `uint64` seems wasteful, shall we have it as `uint8` instead?

Reply:
Yes, I will change it also in Request and RetrieveRequestMsg to uint8, there is no need for more.